TODOs.txt

```diff
@@ -0,0 +1,17 @@
+- Use a uniform API for all async Flight client APIs:
+
+  void Foo(const FlightCallOptions&, shared_ptr<ReadListener<T>> listener);
+
+  - shared_ptr, or raw pointer? (gRPC uses the latter...)
+  - How does this look for writes?
+  - Maybe we always call AddHold and RemoveHold? (It's unclear to me
+    why you'd want to read/write after the stream ends...)
+
+- StopToken needs to be able to run a callback, so that we can cancel gRPC calls.
```

Comment:
Yes, I think there's an issue already open for this? It should be doable now that we use a helper thread for signal-based cancellation.

The main question is the API: do we want to ensure that callbacks are automatically removed at the end of a scope? Something else?

Reply:
I don't think there's a scope that would work, so we would have to explicitly deregister it before calling OnFinish.

Reply:
Or I suppose if we make it the caller's responsibility, it can be deregistered when the listener itself is cleaned up.

Reply:
Ah. 

So frankly, I don't intend to tackle it here, so I'd rather wait and see how things look once it's wired up. (My thought, though, was that if you provide a StopToken in the CallOptions - which is already possible - we should attach and detach the callback at appropriate times.)